shopfloor_reception/services/reception.py

```diff
@@ -1237,6 +1262,11 @@ def _post_shopfloor_created_line(self, selected_line):
         )
 
     def _auto_post_line(self, selected_line):
+        # If user only processed 1/5 and is the only one working on the move,
+        # then selected_line is the only one related to this move.
+        # In such case, we must ensure there's another move line with the remaining
+        # quantity to do, so selected_line is extracted in a new move as expected.
+        selected_line._split_partial_quantity()
         new_move = selected_line.move_id.split_other_move_lines(
```

Comment:
If some other move lines have a product_uom_qty, you need to modify them so that the new_move qty done is not smaller than the remaining move lines product_uom_qty. To prevent this complexity, it is easier if all move lines have a product_uom_qty of 0.
```suggestion
        selected_line.move_id.move_line_ids.filtered("product_uom_qty").product_uom_qty = 0
        new_move = selected_line.move_id.split_other_move_lines(
```

Reply:
We decided to revert the change where I introduced the "align_qties" method, and to modify the product_uom_qty in the response instead, in order to not alter a move line the user is not working on.
I'm, we should set the qty in `_scan_line__find_or_create_line` every time, whether we create a new line or find one.

Reply:
Drop the call to split_other_move_lines and do sth like:
```
if selected_line.qty_done < selected_line.move_id.qty_todo:
            split_move_vals = self._split(selected_line.qty_done)
            split_move = self.create(split_move_vals)
            split_move.move_line_ids = to_move
            split_move._action_confirm(merge=False)
            split_move._recompute_state()
            split_move._action_assign()
            self._recompute_state()
            return split_move
```

Reply:
@jbaudoux can you make another pass please?